app/src/main/java/com/example/atheneum/activities/BookInfoActivity.java

```diff
@@ -370,6 +408,96 @@ public void deleteBook(){
         }
     }
 
+    /**
+     * Attempt to get information from goodreads
+     */
+    public void getGoodreadsReviewInfo(long isbn) {
+        // show error instead if there is no internet connection
+        ConnectionChecker connectionChecker = new ConnectionChecker(this);
```

Comment:
I don't know if this is possible or not (and it's super minor tbh) but is it possible to try to show the goodreads review info as soon as the internet comes online again. Like if i click on a book while offline and i start wifi again, it currently doesn't show the reviews until i go back to the main activity and click the book again.

Reply:
Hmm I can take a look.

Reply:
Okay, It looks kinda like a pain to do. 
Going to merge this first and maybe make a separate PR to add the network switch check.